charts/nr-k8s-otel-collector/templates/daemonset.yaml

```diff
@@ -0,0 +1,98 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ include "newrelic.common.naming.fullname" . }}-daemonset
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "newrelic.common.labels" . | nindent 4 }}
+spec: 
+  selector:
+    matchLabels:
+      {{- include "newrelic.common.labels.selectorLabels" . | nindent 6 }}
+      component: daemonset
+  template:
+    metadata:
+      labels:
+        {{- include "newrelic.common.labels.selectorLabels" . | nindent 8 }}
+        component: daemonset
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/daemonset-configmap.yaml") . | sha256sum }}
+    spec:
+      serviceAccountName: {{ include "newrelic.common.serviceAccount.name" . }}
+      containers: 
+        - name: otel-collector-daemonset
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          command:
+            - /otelcol-contrib
+            - --config=/conf/otel-collector-config.yaml
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
```

Comment:
The reason behind the `CHART-TEMPLATE` chart was not only to have a testing for `common-library` but also to have a template to create new charts instead of `helm create`.

There are still a lot of features from the common library that are not implemented here: https://github.com/newrelic/helm-charts/blob/162de8eac4822b633ad4e08ff6660035a677c328/library/CHART-TEMPLATE/templates/sidecar-deployment.yaml#L21-L44

Reply:
Gotcha. We can add the missing common library features in a follow-up PR